FWCore/Framework/src/StreamSchedule.cc

```diff
@@ -158,6 +158,68 @@ namespace edm {
       }
       return std::pair(beg + 1, std::prev(modnames.end()));
     }
+
+    std::optional<std::string> findBestMatchingAlias(
+        std::unordered_multimap<std::string, edm::BranchDescription const*> const& conditionalModuleBranches,
+        std::unordered_multimap<std::string, StreamSchedule::AliasInfo> const& aliasMap,
+        std::string const& productModuleLabel,
+        ConsumesInfo const& consumesInfo) {
+      std::optional<std::string> best;
+      int wildcardsInBest;
+
+      auto updateBest = [&best, &wildcardsInBest, &consumesInfo, &productModuleLabel](
+                            std::string const& label, bool instanceIsWildcard, bool typeIsWildcard) {
+        int const wildcards = static_cast<int>(instanceIsWildcard) + static_cast<int>(typeIsWildcard);
+        if (wildcards == 0) {
+          return true;
+        }
+        if (not best or wildcards < wildcardsInBest) {
+          best = label;
+          wildcardsInBest = wildcards;
+        } else if (best and *best != label and wildcardsInBest == wildcards) {
+          throw Exception(errors::UnimplementedFeature)
```

Comment:
So if you had 3 choices, two with only one wild card options and the third with a wild card options, you may or may not cause this exception depending on what order they were specified in?

Reply:
I'm not sure what you meant with
> and the third with a wild card options

but in case "without any wild card options" I think you're right, whether or not the exception is thrown depends on the order :( _Maybe_ it would be sufficient to remove the shortcut `return true` for the case if a matching no-wildcard alias is found and process them all every time?

Reply:
> _Maybe_ it would be sufficient to remove the shortcut `return true` for the case if a matching no-wildcard alias is found and process them all every time?

It's not sufficient because the no-wildcard alias could be found between two one-wildcard cases. I suppose I'd have to keep track separately if the best case found so far has an ambiguity, and throw the exception if the identified best case has the ambiguity. (that would also allow keeping the short-circuit for no-wildcard alias).

Reply:
> I suppose I'd have to keep track separately if the best case found so far has an ambiguity, and throw the exception if the identified best case has the ambiguity. (that would also allow keeping the short-circuit for no-wildcard alias).

Done